awx/main/management/commands/cleanup_images.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2015 Ansible, Inc.
+# All Rights Reserved.
+
+# Python
+import subprocess
+import logging
+import json
+
+# Django
+from django.core.management.base import BaseCommand, CommandError
+from django.conf import settings
+
+# AWX
+from awx.main.models import ExecutionEnvironment
+
+
+class Command(BaseCommand):
+    """
+    Management command to cleanup unused execution environment images.
+    """
+
+    help = 'Remove unused execution environment images'
+
+    def init_logging(self):
+        log_levels = dict(enumerate([logging.ERROR, logging.INFO, logging.DEBUG, 0]))
+        self.logger = logging.getLogger('awx.main.commands.cleanup_images')
+        self.logger.setLevel(log_levels.get(self.verbosity, 0))
+        handler = logging.StreamHandler()
+        handler.setFormatter(logging.Formatter('%(message)s'))
+        self.logger.addHandler(handler)
+        self.logger.propagate = False
+
+    def add_arguments(self, parser):
+        parser.add_argument('--dry-run', dest='dry_run', action='store_true', default=False, help='Dry run mode (show items that would ' 'be removed)')
+
+    def delete_images(self, images_json):
+        if self.dry_run:
+            delete_prefix = "Would delete"
+        else:
+            delete_prefix = "Deleting"
+        for e in images_json:
+            if 'Names' in e:
+                image_names = e['Names']
+            else:
+                image_names = [e["Id"]]
+            image_size = e['Size'] / 1e6
+            for i in image_names:
+                if i not in self.images_in_use and i not in self.deleted:
+                    self.deleted.append(i)
+                    self.logger.info(f"{delete_prefix} {i}: {image_size:.0f} MB")
+                    if not self.dry_run:
+                        subprocess.run(['podman', 'rmi', i, '-f'], stdout=subprocess.DEVNULL)
```

Comment:
_Just in case_ something goes wrong, it might be nice to have e.g.,

```python
process = subprocess.run(...)
if process.returncode != 0:
   logging_something()
```

Reply:
fixed